src/adagrams.js

```diff
@@ -1,15 +1,107 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
-};
+  const LETTER_POOL = {
+    A: 9, B: 2, C: 2, D: 4, E: 12, F: 2, G: 3, H: 2,
+    I: 9, J: 1, K: 1, L: 4, M: 2, N: 6, O: 8, P: 2,
+    Q: 1, R: 6, S: 4, T: 6, U: 4, V: 2, W: 2, X: 1,
+    Y: 2, Z: 1
+  };
+  
+  const MAX_HAND_COUNT = 10;
+  let availableLetters = [];
+
+  for (let letter in LETTER_POOL) {
+    let originalLetterCount = LETTER_POOL[letter];
+    for (let i = 0; i < originalLetterCount; i++) {
+      availableLetters.push(letter);
+    }
+  }
+
+  let currentHand = [];
+
+  while (currentHand.length < MAX_HAND_COUNT) {
+    const randomIndex = Math.floor(Math.random() * availableLetters.length);
+    const letter = availableLetters[randomIndex];
+
+    currentHand.push(letter);
+    availableLetters.splice(randomIndex, 1);
+  }
+
+  return currentHand;
+}
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  const word = input.toUpperCase();
+
+  for (const inputLetter of word) {
+    const letterFreqInLetterBank = lettersInHand.filter((letter) => letter === inputLetter).length;
+    const letterFreqInWord = word.split('').filter((letter) => letter === inputLetter).length;
+
+    if (letterFreqInWord > letterFreqInLetterBank) {
+      return false;
+    }
```

Comment:
The comment found above could apply here as well. Again, these are constant sized inputs but consider if they weren't.